test/task.cc

```diff
@@ -336,9 +423,7 @@ TEST(Task, FailContinuation) {
               std::tuple<>>,
           std::tuple<std::runtime_error>>);
 
-  EXPECT_THAT(
-      [&]() { std::rethrow_exception(result); },
-      ThrowsMessage<std::runtime_error>(StrEq("error")));
+  EXPECT_STREQ(result.value().what(), "error");
```

Comment:
Let's just do this in the lambda above rather than copying it outside via a `std::optional`.